File: benchmarks/json_parse.go
package main

import (
	"encoding/json"
	"fmt"
)

type User struct {
	Name string `json:"name"`
	Age  int    `json:"age"`
}

type Data struct {
	Users []User `json:"users"`
	Count int    `json:"count"`
}

func main() {
	data := []byte(`{"users": [{"name": "Alice", "age": 30}, {"name": "Bob", "age": 25}], "count": 2}`)

	for i := 0; i < 10000; i++ {
		var parsed Data
		json.Unmarshal(data, &parsed)
		_ = parsed.Users[0].Name
	}

	fmt.Println("JSON parsed 10000 times")
}

File: benchmarks/json_parse.py
#!/usr/bin/env python3
"""JSON parsing benchmark."""

import json

data = (
    '{"users": [{"name": "Alice", "age": 30}, {"name": "Bob", "age": 25}], "count": 2}'
)

for _ in range(10000):
    parsed = json.loads(data)
    _ = parsed["users"][0]["name"]

print("JSON parsed 10000 times")
